src/Console/Command/CustomRuleCommand.php

```diff
@@ -126,9 +132,144 @@ static function (string $answer): string {
             }
         }
 
+        // 3. update phpunit.xml(.dist) to include rector test suite
+        $this->setupRectorTestSuite($currentDirectory);
+
         return Command::SUCCESS;
     }
 
+    private function setupRectorTestSuite(string $currentDirectory): void
+    {
+        if (! extension_loaded('dom')) {
+            $this->symfonyStyle->warning(
+                'The "dom" extension is not loaded. Rector could not add the rector test suite to phpunit.xml'
+            );
+
+            return;
+        }
+
+        $phpunitXmlExists = file_exists($currentDirectory . '/phpunit.xml');
+        $phpunitXmlDistExists = file_exists($currentDirectory . '/phpunit.xml.dist');
+
+        if (! $phpunitXmlExists && ! $phpunitXmlDistExists) {
+            $this->symfonyStyle->warning(
+                'No phpunit.xml or phpunit.xml.dist found. Rector could not add the rector test suite to it'
+            );
+
+            return;
+        }
+
+        $phpunitFile = $phpunitXmlExists ? 'phpunit.xml' : 'phpunit.xml.dist';
+
+        $phpunitFilePath = $currentDirectory . '/' . $phpunitFile;
+
+        $domDocument = new DOMDocument('1.0');
+        $domDocument->preserveWhiteSpace = false;
+        $domDocument->formatOutput = true;
+        $domDocument->loadXML(FileSystem::read($phpunitFilePath));
+
+        if ($this->hasRectorTestSuite($domDocument)) {
+            $this->symfonyStyle->success(
+                'The rector test suite already exists in ' . $phpunitFilePath . ". No changes were made.\n You can run the rector tests by running: phpunit --testsuite rector"
+            );
+
+            return;
+        }
+
+        $phpunitXML = $this->updatePHPUnitXMLFile($domDocument, $phpunitFilePath);
+
+        FileSystem::write($phpunitFilePath, $phpunitXML);
```

Comment:
pass `null` to 3rd arg to ensure keep permission

```suggestion
FileSystem::write($phpunitFilePath, $phpunitXML, null);
```